python/common/org/python/types/Str.java

```diff
@@ -66,14 +66,76 @@ public Str(org.python.Object[] args, java.util.Map<java.lang.String, org.python.
     //     throw new org.python.exceptions.NotImplementedError("__init__() has not been implemented.");
     // }
 
+    private static boolean isCharPrintable(char c) {
+        // ASCII non-printable
+        if ((int) c <= 0x1f || (int) c >= 0x7f && (int) c <= 0xa0 || (int) c == 0xad) {
+            return false;
+        }
+        if ((int) c == 0x2029) {
+            return false;
+        }
+        if (Character.isISOControl(c)) {
+            return false;
+        }
+        return true;
+    }
+
     @org.python.Method(
             __doc__ = ""
     )
     public org.python.Object __repr__() {
-        String repr = this.value.replaceAll("\\n", "\\\\n").replaceAll("\\r", "\\\\r");
-        return new org.python.types.Str("'" + repr + "'");
+    /*
+    * Reference: https://www.python.org/dev/peps/pep-3138/#id7
+    * TODO: Need to treat the leading surrogate pair characters
+    */
+        StringBuilder sb = new StringBuilder();
+        boolean has_double_quote = false;
```

Comment:
This is a really minor style point, but Java usually has camelCase variable names instead of snake_case.